### PR TITLE
Added Table Columns

### PR DESCRIPTION
**Pull request by MattCloward**
Added columns to associations_table:
- betaValue
- betaUnit
- betaAnnotation
- ogValueTypes
- numAssociationsFiltered

Added columns to study_table:
- superPopulation
- sex
- pValueAnnotation
- betaAnnotation
- ogValueTypes
- numRSFiltered

Table fixes:
Fixed sexes being set for studies without sexes specified
Removed associations "conditioned on" other SNPs
Re-added associations that have valid RSIDs, but no position listed.
The lastUpdated tibble now updates whenever the associations_table updates instead of all at the end
Fixed ethnicities showing up as "" instead of NA

### CODE REVIEW

File: update_database_scripts/createStudyTable.R [hunk 1]
```diff
@@ -13,10 +13,16 @@
...
+#        "altmetricScore" is the Altmetric score given to the study- it is a measure of the popularity of the study (see altmetric.com for more info)
+#        "ethnicity" is a pipe (|) separated string of ethnicities involved in the study
+#        "superPopulation" is a pipe (|) separated string of super populations involved in the study (based on the 5 populations from 1000 genomes)
 #        "initialSampleSize" is the intitial sample size of the study
 #        "replicationSampleSize" is the replication sample size of the study
+#        "sexes" is a pipe (|) deliminated string containing the sexes listed in the study ("male", "female", or "male|female")
+#        "pValueAnnotations" is a pipe (|) deliminated string containing all the pValueAnnotations in the study
+#        "betaAnnotations" is a pipe (|) deliminated string containing all the betaAnnotations in the study
+#        "ogValueTypes" is a pipe (|) deliminated string containing the study's value types ("OR", "beta", or "OR|beta")
+#        "numRSFiltered" is the number of SNPs filtered from the study (not in the associations table)
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R]: Need to update this name

File: update_database_scripts/createStudyTable.R [hunk 2]
```diff
@@ -141,6 +147,38 @@ getDatabaseTraitName <- function(traitName) {
...
 
+getSuperPop <- function(ethnicity) {
+  if (is.na(ethnicity)) {
+    return(NA_character_)
+  } else {
+    superPopulation <- c()
+    if (str_detect(ethnicity, regex("african", ignore_case = TRUE))) {
+      superPopulation <- c(superPopulation, "African")
+    } 
+    if (str_detect(ethnicity, regex("(?<!african )american", ignore_case = TRUE))) { # match american, but not african american (negatve lookbehind)
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R:158]: that is cool!

File: update_database_scripts/createStudyTable.R [hunk 3]
```diff
@@ -155,7 +193,23 @@ if (is_ebi_reachable()) {
...
+                         "altmetricScore" = double(0), 
+                         "ethnicity" = character(0),
+                         "superPopulation" = character(0),
+                         "initialSampleSize" = numeric(0), 
+                         "replicationSampleSize" = numeric(0), 
+                         "sexes" = character(0), 
+                         "pValueAnnotations" = character(0), 
+                         "betaAnnotations" = character(0), 
+                         "ogValueTypes" = character(0),
+                         "numRSFiltered" = numeric(0),
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R]: Rename

File: update_database_scripts/createStudyTable.R [hunk 4]
```diff
@@ -174,8 +228,13 @@ if (is_ebi_reachable()) {
...
     # only have 2 traits).
     studyIDRows <- group_by(associationsTibble, studyID) %>%
-      mutate(trait = paste0(unique(trait[!is.na(trait)]), collapse = "|"))
-    studyIDRows <- select(arrange(distinct(studyIDRows, studyID, .keep_all = TRUE), studyID), studyID, citation, trait)
+      mutate(trait = paste0(unique(trait[!is.na(trait)]), collapse = "|")) %>%
+      mutate(sexes = paste0(unique(sex[!is.na(sex)]), collapse = "|")) %>%
+      mutate(ogValueTypes = paste0(unique(ogValueTypes[!is.na(ogValueTypes)]), collapse = "|")) %>%
+      mutate(pValueAnnotations = paste0(unique(pValueAnnotation[!is.na(pValueAnnotation)]), collapse = "|")) %>%
+      mutate(betaAnnotations = paste0(unique(betaAnnotation[!is.na(betaAnnotation)]), collapse = "|")) %>%
+      mutate(numRSFiltered = paste0(unique(numRSFiltered[!is.na(numRSFiltered)]), collapse = "|"))
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R]: Rename

File: update_database_scripts/createStudyTable.R [hunk 5]
```diff
@@ -209,16 +268,24 @@ if (is_ebi_reachable()) {
...
         lastUpdated <- as.character(lastUpdatedTibble[lastUpdatedTibble$studyID == studyID,]$lastUpdated)
         
         initialSampleSize <- SumNumsFromString(rawStudyData[["initial_sample_size"]])
         replicationSampleSize <- SumNumsFromString(rawStudyData[["replication_sample_size"]])
         
+        sexes <- studyIDRow[["sexes"]]
+        pValueAnnotations <- studyIDRow[["pValueAnnotations"]]
+        betaAnnotations <- studyIDRow[["betaAnnotations"]]
+        ogValueTypes <- studyIDRow[["ogValueTypes"]]
+        numRSFiltered <- strtoi(studyIDRow[["numRSFiltered"]])
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R]: rename

File: update_database_scripts/createStudyTable.R [hunk 6]
```diff
@@ -230,8 +297,14 @@ if (is_ebi_reachable()) {
...
                               altmetricScore = altmetricScore,
                               ethnicity = ethnicity,
+                              superPopulation = superPopulation,
                               initialSampleSize = initialSampleSize,
                               replicationSampleSize = replicationSampleSize,
+                              sexes = sexes,
+                              pValueAnnotations = pValueAnnotations,
+                              betaAnnotations = betaAnnotations,
+                              ogValueTypes = ogValueTypes,
+                              numRSFiltered = numRSFiltered,
```
Comment (mpage21) [inline update_database_scripts/createStudyTable.R]: rename

File: update_database_scripts/unpackDatabaseCommandLine.R [hunk 7]
```diff
@@ -444,16 +530,23 @@ if (is_ebi_reachable()) {
...
         add_column(citation = citation) %>%
         add_column(studyID = studyID, .after = "citation") %>%
-        add_column(sex = getSexesFromDescriptions(master_associations[["pvalue_description"]])) %>%
+        add_column(sex = getSexesFromDescriptions(master_associations[["pvalue_description"]], master_associations[["beta_description"]])) %>%
+        add_column(betaValue = getBeta(master_associations[["beta_number"]], master_associations[["beta_direction"]]), .after = "upperCI") %>%
+        add_column(ogValueTypes = getOGValueTypes(master_associations[["or_per_copy_number"]], master_associations[["beta_number"]]), .after = "betaValue") %>%
         mutate(pvalue_description = tolower(pvalue_description))
-      # filter out SNPs on the X or Y chromosome
-      studyData <- filter(studyData, !startsWith(hg38, "X")&!startsWith(hg38, "Y"))
+      # filter out SNPs on the X or Y chromosome (keeping NAs)
```
Comment (mpage21) [inline update_database_scripts/unpackDatabaseCommandLine.R:561]: Is this line ensuring that when we have a valid rsid, but the chrome:pos is NA we keep it?

File: update_database_scripts/unpackDatabaseCommandLine.R [hunk 8]
```diff
@@ -380,17 +486,21 @@ if (is_ebi_reachable()) {
       # gets the association data associated with the study ID
       associations <- get_associations(study_id = studyID)
       associationsTibble <- associations@associations
-      # filter out blank ORs, invalid ORs, and SNPs associated with other SNPs
-      associationsTibble <- filter(associationsTibble, !is.na(or_per_copy_number)&(or_per_copy_number > -1)&!grepl("condition", pvalue_description)&!grepl("adjusted for rs", pvalue_description))
+      numTotatAssociations <- nrow(associationsTibble)
```
Comment (mpage21) [inline update_database_scripts/unpackDatabaseCommandLine.R:489]: *total

File: update_database_scripts/unpackDatabaseCommandLine.R [hunk 9]
```diff
@@ -438,22 +548,29 @@ if (is_ebi_reachable()) {
...
-      # filter out SNPs on the X or Y chromosome
-      studyData <- filter(studyData, !startsWith(hg38, "X")&!startsWith(hg38, "Y"))
+      # filter out SNPs on the X or Y chromosome (keeping NAs)
+      studyData <- filter(studyData, !startsWith(hg38, "X")&!startsWith(hg38, "Y")|is.na(hg38))
       # check if studyData has enough snps
       if (is.null(checkIfValidDataObj(studyData))) {next}
+      
+      # calculate the total number of SNPS that have been filtered out for the study and add it as a column      
+      numFinalAssociations <- length(unique(studyData[["variant_id"]]))
+      numAssociationsFiltered <- numTotatAssociations - numFinalAssociations
```
Comment (mpage21) [inline update_database_scripts/unpackDatabaseCommandLine.R:568]: *total

**Review by mpage21 — CHANGES_REQUESTED** [review]
Mostly spelling errors

File: update_database_scripts/unpackDatabaseCommandLine.R (hunk 7 above)
Comment (MattCloward) [reply update_database_scripts/unpackDatabaseCommandLine.R:561]: Yes

**Review by mpage21 — APPROVED** [review]
I didn't look too in-depth, but it looks good!

**Pull request merged**